README.md

```diff
@@ -28,3 +28,46 @@ export default {
 ```
 
 Check the examples/minimal folder for a complete example.
+
+## Gotchas and/or Limitations
+
+### Provide unique names for Modules and Tests
+
+Names (and assertion messages) are used as unique identifiers to build up the test results that @web/test-runner is expecting. Most of the actions to build this data structure are only appending new data. Take for example the following test code:
+
+```
+QUnit.module('Module A', () => {
+  QUnit.test('test 1', assert => {
+    assert.true(true, 'assertion 1')
+  })
+  QUnit.test('test 2', assert => {
+    assert.true(true, 'assertion 2')
+  })
+})
+QUnit.module('Module A', () => {
+  QUnit.test('test 1', assert => {
+    assert.true(true, 'assertion 3')
+  })
+})
+```
+
+This would result in the following summary output:
+
+```
+ Module A [Chrome]
+  test 1 [Chrome]
+    ✓ assertion 1
+    ✓ assertion 3
+  test 2 [Chrome]
+    ✓ assertion 2
+```
```

Comment:
If i skip the assertion description (e.g. assert.true(true)), the test name is printed. Is it really desired to print all assertions from tests in the summary? Or would it make more sense to just mark the test failed and skip the assertions in the summary? the error of the affected assertion can be included anyways. What do you think?

Reply:
It has been a bit of a struggle to find the best mapping to the @web/test-runner data. If we go back to mapping the `QUnit.test` to the [`TestResult`](https://github.com/modernweb-dev/web/blob/f7fcf29cb79e82ad5622665d76da3f6b23d0ef43/packages/test-runner-core/src/test-session/TestSession.ts#L24) then we can only report a single failure at a time. It seemed better to map each assertion inside of a `QUnit.test` to a `TestResult` instead in order to capture all failed assertions within a test.

I'm honestly unsure which compromise is better: one failed assertion reporting per test per a run or reporting on every assertion as individual tests.

Reply:
Maybe It'd be nicer to output a default message if the assertion does not have one instead of the test name. Something like `assertion [index + 1]`.

Reply:
Hmm... thinking about this some more. In my [original implementation](https://github.com/brandonaaron/web-test-runner-qunit/blob/2e81cba5b5ab335a6ca16c0513ce314893a0722b/src/autorun.ts#L121) I treated each assertion failure as a [global error](https://github.com/modernweb-dev/web/blob/f7fcf29cb79e82ad5622665d76da3f6b23d0ef43/packages/test-runner-core/src/test-session/TestSession.ts#L32). With that approach the `QUnit.test` still appropriately maps to a `TestResult` and we'll just skip associating the `TestResultError` and instead push it to the `errors` on the `TestSession`.

Yeah, I think there is a happy blend between this branch and the main branch that gets us where we want to be. I'll try to make some time later today to keep iterating on this.

Reply:
I worked on this a bit just now but unfortunately @web/test-runner does not output the expected/actual diff for the [`TestSession.errors`](https://github.com/modernweb-dev/web/blob/f7fcf29cb79e82ad5622665d76da3f6b23d0ef43/packages/test-runner-core/src/test-session/TestSession.ts#L32).

This is a screen grab of a session where it is showing how it outputs the errors. I also used `console.log` to output what I'm giving @web/test-runner.

<img width="975" alt="Screenshot 2024-02-14 at 8 09 10 AM" src="https://github.com/brandonaaron/web-test-runner-qunit/assets/13718/aa56e7c7-84c3-4195-a4c3-8500f4e40387">

This puts us back to compromising on displaying all assertions or only exposing the first assertion failure in a test. I was initially leaning more towards mapping all QUnit assertions to an @web/test-runner `TestResult`. However, it is a bit simpler to just map the QUnit.test to the `TestResult` and only show the first failing assertion. In the future there could be a config item that would change the behavior to instead show all assertions as initially done in this PR.

Reply:
I went ahead and implemented the change to map each `QUnit.test` to a `TestResult` and only show the first failed assertion.

Reply:
Maybe is miss something completely, but why are you not adding the assertion failures to testResults? The missing diff output when adding them to errors was the reason for my initial issue report ;)

Reply:
I might not be conveying the information very well. While iterating through ideas/solutions I was mentioning that @web/test-runner does not show the actual/expected diff when sending errors as TestSession.errors. However, in the latest commit this does pass along the assertion failure (but only one per test).

Given the following tests:

```
QUnit.test('testing 1', assert => {
  assert.true(true, 'assertion 1')
  assert.true(false, 'assertion 2')
  assert.true(false, 'assertion 3')
  assert.true(true, 'assertion 4')
})
```

The output will be:

<img width="587" alt="Screenshot 2024-02-14 at 8 55 02 AM" src="https://github.com/brandonaaron/web-test-runner-qunit/assets/13718/2c6345ce-93f2-44e4-9a29-02d5f6a4e29b">

Reply:
Hey @rnixx, I wanted to check in to see if you've tried the latest in this PR? I'd appreciate your input on how it works for you.

Reply:
@brandonaaron Sorry for my late reply, I am quite busy these days. I just tried your changes and at a first glance it looks really good now! Just one minor thing. Instead of printing 'Unknown Error' if an assertion description is missing maybe 'Assetion Error' would be more appropriate. Thanks for all the improvements you did!

Reply:
I like the idea of using 'Assertion Error' as the fallback for a missing assertion message and got that added.